calcom/cal.com
# chore: [app-router-migration 8.6] reorganize future pages file structure

## What does this PR do?

- removes Route Groups. This is workaround for the issue where unnecessary chunks are included to the page. See [document1](https://www.notion.so/intuita/Debugging-increased-LCP-on-teams-and-videos-74fabbdb00e94c41815ff0c0eb9610d1?pvs=4#d5ac6402777e4ca49da516c000de4c5e)
 [document2](https://intuita.notion.site/Route-group-issue-4ed00b85fb104266b9581073d5964a29)
 
- adds WithLayout HOC. Replace a lot of copy-paste code in layout files with a single HOC that accepts `getLayout`, `getData` and LegacyPage. 

## Requirement/Documentation
[document1](https://www.notion.so/intuita/Debugging-increased-LCP-on-teams-and-videos-74fabbdb00e94c41815ff0c0eb9610d1?pvs=4#d5ac6402777e4ca49da516c000de4c5e)
[document2](https://intuita.notion.site/Route-group-issue-4ed00b85fb104266b9581073d5964a29)

## Type of change




## How should this be tested?

- Navigate to the page that previously included extra layout chunk (for instance `future/video/no-meeting-found`). Verify that only 1 layout chunk is included. Verify that SettingsLayout is not present in the bundle. 

## Mandatory Tasks
